chrome/content/zotero/xpcom/attachments.js

```diff
@@ -369,25 +380,7 @@ Zotero.Attachments = new function(){
 				// Refetch item to update path
 				attachmentItem.attachmentPath = destFile.path;
 				yield attachmentItem.save(saveOptions);
-			}.bind(this))
-			.catch(function (e) {
-				Zotero.debug(e, 1);
-				
-				// Clean up
-				try {
-					if (tmpDir && tmpDir.exists()) {
-						tmpDir.remove(true);
-					}
-					if (destDir && destDir.exists()) {
-						destDir.remove(true);
-					}
-				}
-				catch (e) {
-					Zotero.debug(e, 1);
-				}
-				
-				throw e;
-			});
```

Comment:
I don't think this is right — this way, if something in the transaction fails, stuff won't get deleted.